src/components/chart/chart.tsx

```diff
@@ -0,0 +1,450 @@
+ * @exampleComponent limel-example-chart-type-scatter
+ * @exampleComponent limel-example-chart-type-area
+ * @exampleComponent limel-example-chart-type-line
+ * @exampleComponent limel-example-chart-type-pie
+ * @exampleComponent limel-example-chart-type-doughnut
+ * @exampleComponent limel-example-chart-type-ring
+ * @exampleComponent limel-example-chart-type-gantt
+ * @exampleComponent limel-example-chart-multi-axis
+ * @exampleComponent limel-example-chart-multi-axis-with-negative-start-values
+ * @exampleComponent limel-example-chart-multi-axis-area-with-negative-start-values
+ * @exampleComponent limel-example-chart-axis-increment
+ * @exampleComponent limel-example-chart-accessibility
+ * @exampleComponent limel-example-chart-styling
+ * @exampleComponent limel-example-chart-creative-styling
+ * @Beta
+ */
+
+@Component({
+    tag: 'limel-chart',
+    shadow: true,
+    styleUrl: 'chart.scss',
+})
+export class Chart {
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop({ reflect: true })
+    public language: Languages = 'en';
+
+    /**
+     * Helps users of assistive technologies to understand
+     * the context of the chart, and what is being displayed.
+     */
+    @Prop({ reflect: true })
+    public accessibleLabel?: string;
+
+    /**
+     * Helps users of assistive technologies to understand
+     * what the items in the chart represent.
+     */
+    @Prop({ reflect: true })
+    public accessibleItemsLabel?: string;
+
+    /**
+     * List of items in the chart,
+     * each representing a data point.
+     */
+    @Prop()
+    public items!: ChartItem[];
+
+    /**
+     * Defines how items are visualized in the chart.
+     */
+    @Prop({ reflect: true })
+    public type?:
+        | 'area'
+        | 'bar'
+        | 'doughnut'
+        | 'line'
+        | 'pie'
+        | 'ring'
+        | 'scatter'
+        | 'stacked-bar' = 'stacked-bar';
+
+    /**
+     * Defines whether the chart is intended to be displayed wide or tall.
+     * Does not have any effect on chart types which generate circular forms.
+     */
+    @Prop({ reflect: true })
+    public orientation?: 'landscape' | 'portrait' = 'landscape';
+
+    /**
+     * Specifies the range that items' values could be in.
+     * This is used in calculation of the size of the items in the chart.
+     * When not provided, the sum of all values in the items will be considered as the range.
+     */
+    @Prop({ reflect: true })
+    public maxValue?: number;
+
+    /**
+     * Specifies the increment for the axis lines.
+     */
+    @Prop({ reflect: true })
+    public axisIncrement?: number = DEFAULT_AXIS_INCREMENT;
+
+    private rangeData: {
+        minRange: number;
+        maxRange: number;
+        totalRange: number;
+    };
+
+    private cachedRange: {
+        minRange: number;
+        maxRange: number;
+        totalRange: number;
+    } | null = null;
```

Comment:
After the component is rendered, new items might get added to the dataset, which requires recalculations to redraw the items in the chart. Also the axis lines increments may change after the render. 

However, the overlord thought that:

> The function `calculateRange()` is being triggered whenever `axisIncrement`, `items`, or `maxValue` changes. Since this involves iterating through the items and performing calculations, consider memoizing the results using something like a cache to prevent recalculating unnecessarily if the values haven’t changed.

Reply:
Yes, I'm not arguing against the calculation itself. However, `rangeData` is only ever set to a new value inside `recalculateRangeData`, like so
```ts
    private recalculateRangeData() {
        this.rangeData = this.calculateRange();
    }
```
and `cachedRange` is only ever set to a new value inside `calculateRange`, which is then returned, so they will always be the same. I don't see why we need both

Reply:
🙈